vaadin-checkbox-flow-parent/vaadin-checkbox-testbench/src/main/java/com/vaadin/flow/component/checkbox/testbench/CheckboxGroupElement.java

```diff
@@ -0,0 +1,140 @@
+     *
+     * @return a list of all checkboxes in this group
+     */
+    public List<CheckboxElement> getCheckboxes() {
+        return $(CheckboxElement.class).all();
+    }
+
+    /**
+     * Selects the first checkbox matching the given text. If it is selected
+     * does nothing.
+     *
+     * @param text
+     *            the text of the option to select
+     */
+    public void selectByText(String text) {
+        Optional<CheckboxElement> checkbox = getCheckboxByText(text);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No item with text '" + text + "' found");
+        }
+
+        checkbox.get().setChecked(true);
+    }
+
+    /**
+     * Unselects the first checkbox matching the given text. If it is unslected
+     * does nothing.
+     *
+     * @param text
+     *            the text of the option to select
+     */
+    public void unselectByText(String text) {
+        Optional<CheckboxElement> checkbox = getCheckboxByText(text);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No item with text '" + text + "' found");
+        }
+
+        checkbox.get().setChecked(false);
+    }
+
+    public List<String> getSelectedTexts() {
+        Stream<CheckboxElement> button = getSelectedCheckboxes();
+        return button.map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the selected checkboxes.
+     *
+     * @return an stream with the selected checkboxs elements.
+     */
+    private Stream<CheckboxElement> getSelectedCheckboxes() {
+        return getCheckboxByChecked(true);
+    }
+
+    /**
+     * Gets the checkbox which is part of this group and has the given visible
+     * text.
+     *
+     * @param text
+     *            the text to search for
+     * @return an optional with the checkbox element or an empty optional if no
+     *         match was found
+     */
+    private Optional<CheckboxElement> getCheckboxByText(String text) {
+        if (text == null) {
+            return Optional.empty();
+        }
+        return getCheckboxes().stream()
+                .filter(checkbox -> text.equals(checkbox.getLabel()))
+                .findFirst();
+    }
+
+    /**
+     * Gets the stream of checkboxes which is part of this group and has the
+     * given value.
+     *
+     * @param value
+     *            the value to search for
+     * @return a stream with the checkbox elements
+     */
+    private Stream<CheckboxElement> getCheckboxByChecked(boolean value) {
+        return getCheckboxes().stream()
+                .filter(checkbox -> checkbox.isChecked());
+    }
+
+    /**
+     * Gets the slotted error message component for the element.
+     *
+     * @return the slotted component or {@code null} if there is no component
+     */
+    public TestBenchElement getErrorMessageComponent() {
```

Comment:
nit: I would rather introduce a shared interface later 🤔

Reply:
I was thinking about that too as probably our elements are not consistent on this. But it will separate PR.